### ISSUE TITLE
Feature request: New action(s) - click/tap items in other apps

### ISSUE THREAD

**Issue by sudomain** [sds100/KeyMapper#257]
Great app. I use a closed source app (Automate) for custom keyboard shortcuts, but if KeyMapper implements this feature I can replace Automate:

Action(s) that allows for clicking/tapping items in other apps. There can be a button to record the interaction in the other app so KeyMapper knows what to click. This will be good for apps that use Android provided buttons, switches, and menus. To implement this, I'm guessing that while KeyMapper is running in the background it will need a way to keep track of the current app activity so that custom shortcuts wont bleed between apps.

Another idea would be if KeyMapper fails to detect any clicking actions while recording the interaction, it could fall back on clicking a user designated X,Y  coordinate on the screen. This might be useful for apps such as games that don't use Android provided buttons, switches, and drop menus.

**Comment by sds100** [comment]
- have two options: coordinate or accessibility node. Coordinate will just tap the coordinate and accessibility node will click the node.

**Comment by sudomain** [comment]
Thank you for looking into this feature request, as well as the `tap coordinates` action (it's very cool). How feasible would an action that clicks a user provided/cropped `.png` file of an UI element be? E.g. the user gives Key Mapper a cropped screenshot of a button to tap, Key Mapper searches for it on the screen (assuming it has the accessibility service), and if found Key Mapper taps the center of it?

**Comment by sds100** [comment]
It is technically possible (not sure on the latest Android versions) but there is no elegant way to do it. Lets say the screen is just a picture, finding another picture of random resolution and accuracy in another picture isn't trivial. What is the use case for providing a cropped .png of a UI element? If the UI element is a standard UI element that subclasses View.java then the accessibility can find it with the accessibility nodes API thing.

**Comment by sudomain** [comment]
Sorry I was unclear in my last reply, I thought "UI element" was a generic term rather than specific to Android. Some  usecases I can think of that don't involve Android UI elements are tapping a button on a webpage (e.g. GH notifications) or a VNC client connected to a remote host. A .png search feature could be used as a fallback when a normal Android UI isn't being used

**Comment by sds100** [comment]
Yeah, UI element is a generic term. It is up to the app developers to make their custom Views/UI elements behave nicely with accessibility services. The standard views already behave nicely like Buttons and RecyclerViews already do. Okay, there are useful use-cases for this but I am very unlikely to pursue it since it isn't trivial and there isn't much demand for it.

**Comment by sudomain** [comment]
I thought I'd throw the idea out there. Thank you for the  work you put into this great app.

**Comment by sds100** [comment]
Yup, thanks for the ideas

**Comment by sds100** [comment]
Hi @sudomain @pixel-shock,

Since 3.0 has rewritten a lot of the UI I want/need to redesign what was done in the PR. What do you guys think of the following? The idea is generally the same but a good way to reduce the amount of stuff recorded is to filter by the interaction type in the list.

![Image](https://github.com/user-attachments/assets/6c18e2e2-cae6-4372-9cda-d2cf0f56c84a)

And a cool piece of software... https://github.com/jwlilly/Android-Accessibility-Inspector-App

**Comment by githubsterer** [comment]
Even though you left me out :-( I would say AWESOME!

**Comment by sudomain** [comment]
That looks great

**Comment by jambl3r** [comment]
Oh nice, I was searching for this issue earlier and couldn't find it. Looking great, will discuss with you later @sds100

**Comment by pixel-shock** [comment]
> Hi @sudomain @pixel-shock,
> 
> Since 3.0 has rewritten a lot of the UI I want/need to redesign what was done in the PR. What do you guys think of the following? The idea is generally the same but a good way to reduce the amount of stuff recorded is to filter by the interaction type in the list.
> 
> ![Image](https://github.com/user-attachments/assets/6c18e2e2-cae6-4372-9cda-d2cf0f56c84a)
> 
> And a cool piece of software... https://github.com/jwlilly/Android-Accessibility-Inspector-App

A clear "yessss" from my side 🙌

**Comment by githubsterer** [comment]
Everybody's loving it!

**Comment by sds100** [comment]
The latest mock up for those that are interested. Thank you jambl3r. I will start working on this now.

![Image](https://github.com/user-attachments/assets/5c88a09c-b885-4332-a354-1a30636aedb4)

**Comment by githubsterer** [comment]
Looks very exciting. Please also don't forget the feature to allow multiple actions through a button pop-up list, which I am extremely eager about.

**Comment by sds100** [comment]
implemented
